java/src/main/native/src/map_lookup.cu

```diff
@@ -192,6 +197,10 @@ std::unique_ptr<column> map_lookup(column_view const &map_column, string_scalar
                         get_gather_map_for_map_values<true>(map_column, lookup_key, stream, mr) :
                         get_gather_map_for_map_values<false>(map_column, lookup_key, stream, mr);
 
+  if (gather_map->size() == 0) {
```

Comment:
The gather map being returned will be the same size as the input, because every row in the input has a corresponding row in the output (even if it ends up being null).  It's simpler and faster to check the size before generating the gather map and just return the empty column from the beginning.  That way the gather map generation code doesn't have to check as well.